Microsoft.Toolkit.HighPerformance/Extensions/BoolExtensions.cs

```diff
@@ -24,5 +24,43 @@ public static int ToInt(this bool flag)
         {
             return Unsafe.As<bool, byte>(ref flag);
         }
+
+        /// <summary>
+        /// Converts the given <see cref="bool"/> value to an <see cref="int"/> mask with
+        /// all bits representing the value of the input flag (either 0xFFFFFFFF or 0x00000000).
+        /// </summary>
+        /// <param name="flag">The input value to convert.</param>
+        /// <returns>0xFFFFFFFF if <paramref name="flag"/> is <see langword="true"/>, 0x00000000 otherwise.</returns>
+        /// <remarks>This method does not contain branching instructions.</remarks>
+        [Pure]
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static int ToBitwiseMask32(this bool flag)
+        {
+            byte rangeFlag = Unsafe.As<bool, byte>(ref flag);
```

Comment:
This is potentially buggy. `bool` can be `0` or `not 0`, so if I give a `bool` that is `2`, it will become `0xFFFF_FFFE`

Reply:
Right, I kinda implied that the `bool` values were assumed to be coming from valid CLR ops and not from unsafe/interop scenarios. I just found [this old issue](https://github.com/dotnet/roslyn/issues/24652) you opened a while back about this, that was interesting to read! I'll add some remarks about using this extensions just over `bool` values being the result of comparisons (since `ceq`, `cgt` and other ops are guaranteed to just push `1` or `0`), or `default(bool)`. Thanks for spotting this! 😊

Reply:
Interop actually normalizes `bool`, but you can use `LayoutKind.Explicit` or other tricks (like `Unsafe.As`) to get a `bool` with a non 0/1 value.

Reply:
TIL about interop handling that on its own, I didn't know that!
I'll call out those other two scenarios individually then, thanks!